var/spack/repos/builtin/packages/ginkgo/package.py

```diff
@@ -44,6 +44,9 @@ class Ginkgo(CMakePackage, CudaPackage, ROCmPackage):
         values=("Debug", "Release"),
     )
 
+    # make sure Debug mode is not used when optimizing
+    conflicts("build_type=Debug", when="+full_optimizations", msg="Use Release mode when optimizing")
```

Comment:
I don't think that this change is needed or wanted for our package. 

Our `full_optimizations` means something independent from the `build_type`. In detail, it uses templates to generate much more tailored kernels for some preconditioner algorithms working with small dimensions (namely, Block-Jacobi), significantly increasing compile time at the same time.

If I understand the purpose of this PR, simply setting `build_type=RelWithDebInfo` or `build_type=Release` would be sufficient.